proposals/array_concat.json

```diff
@@ -1,7 +1,7 @@
 {
     "id": "array_concat",
     "summary": "Merge two arrays",
-    "description": "Concatenates two arrays into a single array by appending the second array to the first array. Array labels get discarded from both arrays before merging.",
+    "description": "Concatenates two arrays into a single array by appending the second array to the first array. Array labels are kept only if both given arrays are labeled. Otherwise, the labels get discarded from both arrays.",
```

Comment:
what about common labels?
- throw error?
- change conflicting labels silently?
- discard all labels?

add a parameter for this?

Reply:
Good question. I guess throwing an error would be the easiest solution...
If that's not good, we could discard them.

Changing them silently (to what?) sounds like the worst desirable solution.

Reply:
Throwing an error would also be my first choice, but it might not be the most user friendly option.

Discarding all labels might be even more frustrating for end user compared to just changing the conflicting ones.

 > Changing them silently (to what?) 

You could take the behavior from array_append "the array index as string is used as the label.". 
Another option is appending a number to get a unique name, like file managers sometimes do when you copy a file to folder and the filename is already taken.

Reply:
> You could take the behavior from array_append "the array index as string is used as the label.".

Which could - in theory - also exist and then you need yet another behavior for it.

> Another option is appending a number to get a unique name, like file managers sometimes do when you copy a file to folder and the filename is already taken.

Same as above and you this also seems to be pretty difficult to explain so that users can reliably predict the labels.

So for example, if you want to merge the following labels simply by appending integers (starting at 1), what would you get?
["1", "2", "11", "12"] and ["1"]
What should/would the result be for the label in the second array? "111", "13", "3"?

I think I'd stick with an error until people come up with a need for the renaming behavior or back-end want to implement it. I think we could also add such behavior later in a non-breaking way as adding functionality for something that would usually error doesn't seem very breaking to me. 😅

Reply:
> Same as above and you this also seems to be pretty difficult to explain so that users can reliably predict the labels.

Well I mean if you want to add `file.txt` to a folder, and it already exists, and so does `file 2.txt`, you keep incrementing than number till you have a file name that does not yet exist, be it `file 99.txt` or file 1235.txt`, or ...
The idea is that users might prefer a non-predicable descriptive name over a predictable non-descriptive name, or no name at all.

That being said, I agree that we just could start with throwing an error and improve on that once it becomes an issue

Reply:
I've added the corresponding description and exception to both processes.